editor/actions.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+export function savePost( dispatch, post ) {
+	const isNew = ! post.id;
+	dispatch( {
+		type: 'POST_UPDATE_REQUEST',
+		post,
+		isNew,
+	} );
+	new wp.api.models.Post( post ).save().done( ( newPost ) => {
+		dispatch( {
+			type: 'POST_UPDATE_REQUEST_SUCCESS',
+			post: newPost,
+			isNew,
+		} );
+		if ( isNew && window.history.replaceState ) {
+			window.history.replaceState(
+				{},
+				'Post ' + newPost.id,
+				window.location.href.replace( /&post_id=[^&]*$/, '' )
```

Comment:
Don't know how much we care to address this here, but these are very specific assumptions about URLs where the editor would be loaded. Also breaks down on some valid URLs:

```js
'/admin.php?page=gutenberg&post_id=5'.replace( /&post_id=[^&]*$/, '' )
// Correct: "/admin.php?page=gutenberg"

'/admin.php?post_id=5&page=gutenberg'.replace( /&post_id=[^&]*$/, '' )
// Incorrect: "/admin.php?post_id=5&page=gutenberg"

'/admin.php?page=gutenberg&post_id=5&foo=1'.replace( /&post_id=[^&]*$/, '' )
// Incorrect: "/admin.php?page=gutenberg&post_id=5&foo=1"
```

In a final solution we'll likely want to consider a querystring / URL parser & serializer, such as:

- https://nodejs.org/api/querystring.html
- https://nodejs.org/api/url.html
- https://www.npmjs.com/package/querystringify